docs/DeveloperGuide.md

```diff
@@ -921,36 +921,242 @@ testers are expected to do more *exploratory* testing.
+
+    2. Test case: `ssort name`<br>
+       Expected: Status message responds that stalls has been sorted by name (A-Z). GUI updates to show sorted stalls.
+
+    3. Test case: `ssort reversedname`<br>
+       Expected: Status message responds that stalls has been sorted by name (Z-A). GUI updates to show sorted stalls.
+   
+    4. Test case: `ssort address`<br>
+       Expected: Stalls are not sorted. Error details shown in the status message. GUI does not update.
+
+    5. Other incorrect stall sort commands to try: `ssort`, `ssort abc`<br>
+       Expected: Similar to previous.
+
+### Sort reviews
+
+1. Sorting reviews by stall name, date or rating
+
+    1. Prerequisites: Multiple reviews in the list.
+
+    2. Test case: `rsort name`<br>
+       Expected: Status message responds that reviews has been sorted by name (A-Z). GUI updates to show sorted reviews.
+
+    3. Test case: `rsort reversedname`<br>
+       Expected: Status message responds that reviews has been sorted by name (Z-A). GUI updates to show sorted reviews.
+   
+    4. Test case: `rsort date`<br>
+       Expected: Status message responds that reviews has been sorted by date (Oldest to Newest). GUI updates to show sorted reviews.
+
+    5. Test case: `rsort reverseddate`<br>
+       Expected: Status message responds that reviews has been sorted by date (Newest to Oldest). GUI updates to show sorted reviews.
+   
+    6. Test case: `rsort rating`<br>
+       Expected: Status message responds that reviews has been sorted by rate (Lowest to Highest). GUI updates to show sorted reviews.
+
+    7. Test case: `rsort reversedrating`<br>
+       Expected: Status message responds that reviews has been sorted by date (Highest to Lowest). GUI updates to show sorted reviews.
+
+    8. Test case: `rsort content`<br>
+       Expected: Reviews are not sorted. Error details shown in the status message. GUI does not update.
+
+    9. Other incorrect stall sort commands to try: `rsort`, `rsort abc`<br>
+       Expected: Similar to previous.
+
 
 ### Saving data
 
 1. Dealing with missing/corrupted data files
 
-   1. _{explain how to simulate a missing/corrupted file, and the expected behavior}_
-
-1. _{ more test cases …​ }_
+   1. Test case: Simulate data file is missing by deleting `foodwhere.json` in data folder
+      Expected: Upon launching FoodWhere, the GUI display a default stalls and reviews list.
+   2. Test case: Simulate data file goes corrupt by editing a data field to an invalid value while FoodWhere is running
+      Expected: FoodWhere runs normally and do not crash. However, upon closing and re-launching the program, the GUI display an empty stalls and reviews list.
+   
+2. Commands that change data will modify data file 
+   1. Test case: Run any command that affects the data. E.g. `sadd`, `radd`, `sdel`, `rdel` and exit the program with the `exit` command 
+      Expected: Upon launching FoodWhere, the GUI display a stalls and reviews left off before the `exit` command.
```

Comment:
`the GUI displays the stalls and reviews`

Hold on, this doesn't preserve the current sfind filter - I think this can be worded better.

Reply:
Updated, do check if it sounds better now